### ISSUE TITLE
Date Added/Uploaded columns

### ISSUE THREAD

**Issue by redactedscribe** [nicotine-plus/nicotine-plus#2171]
### Describe the solution you'd like
If columns could be sorted and that sort order be remembered across sessions (#2170), then the default sort for some views would need a column. If I'm not mistaken, the Downloads tab is sorted by the time that each download was requested for download. There currently isn't a way to manually sort the Downloads list back into that order since there is no Date Added column. A Date Uploaded would also be useful for the Uploads tab.

Nicotine 3.2.5

**Comment by mathiascode** [comment]
In my opinion, https://github.com/nicotine-plus/nicotine-plus/discussions/1881#discussioncomment-3063191 would make more sense here. We have many columns, and I'd rather not add another that occupies a lot of screen real estate.

**Comment by redactedscribe** [comment]
The natural order on third press is a nice choice. Some counter arguments for dedicated columns:

- It may not be clear what the natural sort order is.
- A column would allow to display the info that natural sort order is using to sort, i.e. the date, instead of it being hidden.
- Not all columns need to be enabled by default. If the user has the space or doesn't mind scrolling horizontally, they can enabled them.

**Comment by slook** [comment]
I agree that the lack of a Date/Time field seems to be non-nonsensical given that natural sort order is chronological (by order of the items in the json file). It feels like some key data is missing. Several users have said that they would find this data useful (but indeed that could be just because there is not currently a method to view the items unsorted).

> I'd rather not add another that occupies a lot of screen real estate.

I also agree that displaying the proposed new column would be overly cluttered and is not necessary for most users, so if the data field containing the DateTime stamp is added then it should be hidden by default. Thus, the third-press to unsort or an Unsorted menuitem command would be desirable in any case.

**Comment by mathiascode** [comment]
This request is blocked until we've ported the transfer list to the new TreeView class.

**Comment by nicksukie** [comment]
Did this ever get added? Another argument for the upload date-time is that people like to monitor which segments of their collection are popular at different points in time. Could be interesting to then later export said data and visualize it, making a showcase of the contribution one has made to the community over time.
